File: src/main/java/ru/mdashlw/hypixel/entities/Rank.java
package ru.mdashlw.hypixel.entities;

/**
 * Rank on the network.
 */
public enum Rank {
    NORMAL("Normal"),
    VIP("VIP"),
    VIP_PLUS("VIP+"),
    MVP("MVP"),
    MVP_PLUS("MVP+"),
    SUPERSTAR("MVP++"),
    YOUTUBER("YouTuber"),
    HELPER("Helper"),
    MODERATOR("Moderator"),
    ADMIN("Administrator");

    private final String localizedName;

    Rank(final String localizedName) {
        this.localizedName = localizedName;
    }

    /**
     * Localized name.
     *
     * @return Localized name.
     */
    public String getLocalizedName() {
        return this.localizedName;
    }
}

File: src/main/kotlin/ru/mdashlw/hypixel/api/enums/Rank.kt
package ru.mdashlw.hypixel.api.enums

enum class Rank(val uncolorizedName: String, val colorizedName: String) {
    ADMIN("[ADMIN] ", "§c[ADMIN] "),
    MODERATOR("[MOD] ", "§2[MOD] "),
    HELPER("[HELPER] ", "§9[HELPER] "),
    YOUTUBER("[YOUTUBE] ", "§c[§fYOUTUBE§c] "),
    SUPERSTAR("[MVP++] ", "§6[MVP§c++§6] "),
    MVP_PLUS("[MVP+] ", "§b[MVP§c+§b] "),
    MVP("[MVP] ", "§b[MVP] "),
    VIP_PLUS("[VIP+] ", "§a[VIP§6+§a] "),
    VIP("[VIP] ", "§a[VIP] "),
    DEFAULT("", "§7");
}
